data/projects/roberta_external_source.md

```diff
@@ -0,0 +1,50 @@
+For children and students, learning to program becomes increasingly accessible
+through programming platforms like the Open Roberta Lab and Code.org, etc..
+While these platforms provide resources and an environment for novices to explore
+and learn, the browser-centric nature of the exercises and toolchain makes it hard to retrieve
+content, learn about proper project structure, and use third party tools that
+are increasingly used by developers, like static code analysis and code generation.
+
+This proposed project is about loading online resources instead of inserting local files
+into the one-click linter on the coala homepage. This will give students the chance
+to check the coding style and other interesting properties of their projects in
+the browser, without jumping through hoops to retrieve their work from a remote
+source, and installing and learning to use coala's command line interface.
+
+The project should allow, for example, for a student to load and check NEPO-programs from the
+Open Robert Lab directly from the browser. This can later be extended to
+allow for other remote sources, like code.org, 5code, or github/bitbucket projects
+to be checked according to the language hints provided by the respective APIs.
+
+Open Roberta is a platform for students of any age to program consumer-grade
+robotics-hardware in a visual manner, using a blockly-like language called `NEPO`.
+Roberta is a long established international project initiated by the Fraunhofer IAIS
+that aims at nurturing interest in STEM and robotics in young women and students.
+Using Open Roberta as an extensible platform instead of proprietary programmers
+specific to certain robotics hardware is an important part of Roberta's main goal
+of accessibility and outreach for young women and students of all ages.
+
+The student will work with members from the Coala team, and members from
+Fraunhofer IAIS and other Roberta-related projects. The student may acquire
+additional certification through Fraunhofer IAIS in Sankt Augustin, Germany.
+
+#### Milestones
+
+##### GSOC 2017 COMMUNITY BONDING (May 4 - May 30)
+
+* Familiarity with the OpenRoberta partner project is established and a method for external access is designed
+* A design goal is reached with the coala project on how to integrate external data sources into the coala web page
```

Comment:
can we have deliverables? They need to be more concrete, esp for community bonding. Otherwise people including mentors just will find excuses and declare the goals "somewhat reached"